src/core/useVideoTexture.tsx

```diff
@@ -1,24 +1,41 @@
 import * as THREE from 'three'
-import { useEffect } from 'react'
+import { useCallback, useEffect, useRef } from 'react'
 import { useThree } from '@react-three/fiber'
-import { suspend, preload, clear } from 'suspend-react'
+import { suspend } from 'suspend-react'
+import Hls, { HlsConfig } from 'hls.js'
 
 interface VideoTextureProps extends HTMLVideoElement {
   unsuspend?: 'canplay' | 'canplaythrough' | 'loadstart' | 'loadedmetadata'
   start?: boolean
 }
 
-export function useVideoTexture(src: string | MediaStream, props?: Partial<VideoTextureProps>) {
-  const { unsuspend, start, crossOrigin, muted, loop, ...rest } = {
+interface HLSConfiguration {
+  hls: HlsConfig
+}
+
+export function useVideoTexture(src: string | MediaStream, props?: Partial<VideoTextureProps | HLSConfiguration>) {
+  const { unsuspend, start, crossOrigin, muted, loop, hls, ...rest } = {
     unsuspend: 'loadedmetadata',
     crossOrigin: 'Anonymous',
     muted: true,
     loop: true,
     start: true,
     playsInline: true,
+    hls: {},
     ...props,
   }
+  const videoRef = useRef<HTMLVideoElement | null>(null)
   const gl = useThree((state) => state.gl)
+
+  const createHLSBinding = useCallback(() => {
+    if (videoRef.current && typeof src === 'string') {
+      var video: HTMLVideoElement = videoRef.current
+      const _hls = new Hls({ ...hls })
+      _hls.attachMedia(video)
+      _hls.loadSource(src)
+    }
+  }, [])
```

Comment:
Should this teardown at any point? I worry this will create a memory leak.

Reply:
This needs to happen after the video element has been created, can we ensure the useLayoutEffect will run after the video has been created? Since the video instance is passed to a ref and not a state variable.

Reply:
Yes, effects won't fire until suspense resolves.

Reply:
Sure I'll try it out and update the PR, thanks!

Reply:
@CodyJasonBennett I've tried it, you can check it here: https://codesandbox.io/p/sandbox/white-smoke-wx5zht?file=%2Fsrc%2FuseHLSVideoTexture.js%3A51%2C1

But it doesn't work. I believe that this:
```
 const _hls = new Hls({ ...hls })
 _hls.attachMedia(video)
 ```
 
 needs to happen before this: `const texture = new THREE.VideoTexture(video)`
 
 if that is true it has to happen inside the promise